### PR TITLE
SAVE_LATER api intial commit

### PR DESCRIPTION
**Pull request by Subhransupanda2000**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/controllers/CartItemStateController.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.example.bookstoreapp.controllers;
+
+import com.example.bookstoreapp.models.CartItemStateUpdate;
+import com.example.bookstoreapp.services.CartItemStateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/cart")
+public class CartItemStateController {
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/controllers/CartItemStateController.java:10]: We don't need new controller. Do it with cart controller

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class CartItemStateServiceImpl implements CartItemStateService {
+    @Autowired
+    private CartItemEntityRepository cartItemEntityRepository;
+    @Autowired
+    private UserContextService userContextService;
+    @Autowired
+    private CartItemServiceImpl cartItemServiceImpl;
+    @Override
+    public CartItemStateUpdate updateStste(Long catalogItemId, CartItemStateUpdate cartItemStateUpdate) {
+        Optional<CartItemEntity> cartItemEntity = cartItemEntityRepository.findById(catalogItemId);
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:24]: Let say John and Alex added same item to cart. If John does save later then the item for Alex also go save later as per this logic.

We should do save later for John and Alex individually

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.example.bookstoreapp.services.impl;
+
+import com.example.bookstoreapp.entities.CartItemEntity;
+import com.example.bookstoreapp.exceptions.AppRuntimeException;
+import com.example.bookstoreapp.models.*;
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:5]: Avoid wildcard imports.

File: src/main/java/com/example/bookstoreapp/services/CartItemStateService.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookstoreapp.services;
+
+import com.example.bookstoreapp.models.CartItemStateUpdate;
+
+public interface CartItemStateService {
+    CartItemStateUpdate updateStste(Long catalogItemId, CartItemStateUpdate cartItemStateUpdate);
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/CartItemStateService.java:6]: Spelling error in method name .

File: src/main/java/com/example/bookstoreapp/services/CartItemStateService.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookstoreapp.services;
+
+import com.example.bookstoreapp.models.CartItemStateUpdate;
+
+public interface CartItemStateService {
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/CartItemStateService.java:5]: We don't need this interface to update stage. We can use criteria query

File: src/main/java/com/example/bookstoreapp/controllers/CartItemStateController.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.example.bookstoreapp.controllers;
+
+import com.example.bookstoreapp.models.CartItemStateUpdate;
+import com.example.bookstoreapp.services.CartItemStateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/controllers/CartItemStateController.java:6]: Avoid wildcard imports

**Review by rkpatra201 — CHANGES_REQUESTED** [review]
Reviewed

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import com.example.bookstoreapp.repositories.CartItemEntityRepository;
+import com.example.bookstoreapp.services.CartItemStateService;
+import com.example.bookstoreapp.services.UserContextService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:13]: A new service class also not required. We can use cart service

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    private CartItemServiceImpl cartItemServiceImpl;
+    @Override
+    public CartItemStateUpdate updateStste(Long catalogItemId, CartItemStateUpdate cartItemStateUpdate) {
+        Optional<CartItemEntity> cartItemEntity = cartItemEntityRepository.findById(catalogItemId);
+        if(cartItemEntity==null)
+        {
+            throw new AppRuntimeException("Item not found");
+        }
+        CartItemEntity updateItem = cartItemStateUpdate.toEntity();
+        cartItemEntityRepository.save(updateItem);
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:30]: Use criteria  query

**Review by rkpatra201 — CHANGES_REQUESTED** [review]
Reviewed

File: src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java (hunk 8 above)
Comment (Subhransupanda2000) [reply src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:30]: should i write criteria query using cartid and state or catalogitemid and state

Comment (rkpatra201) [reply src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java:30]: > should i write criteria query using cartid and state or catalogitemid and state

Depends on you how you will write criteria query.

**Review by rkpatra201 — CHANGES_REQUESTED** [review]
Replied

**Pull request closed without merging**